File: packages/ruby/lib/tesote_sdk/v2/stubs.rb
module TesoteSdk
  module V2
    # Resource clients whose endpoints are documented but not wired in 0.1.0.
    # Each method raises NotImplementedError so the public surface is discoverable
    # while back-compat constraints stay honest.

    class StubBase
      def initialize(transport)
        @transport = transport
      end
    end

    class Transactions < StubBase
      def list_for_account(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V2::Transactions#list_for_account not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V2::Transactions#get not wired in 0.1.0'
      end

      def export(_account_id, _params = {}, opts: {})
        raise NotImplementedError, 'V2::Transactions#export not wired in 0.1.0'
      end

      def sync(_account_id, opts: {})
        raise NotImplementedError, 'V2::Transactions#sync not wired in 0.1.0'
      end

      def bulk(_payload, opts: {})
        raise NotImplementedError, 'V2::Transactions#bulk not wired in 0.1.0'
      end

      def search(_query = {}, opts: {})
        raise NotImplementedError, 'V2::Transactions#search not wired in 0.1.0'
      end
    end

    class SyncSessions < StubBase
      def list(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V2::SyncSessions#list not wired in 0.1.0'
      end

      def get(_account_id, _id, opts: {})
        raise NotImplementedError, 'V2::SyncSessions#get not wired in 0.1.0'
      end
    end

    class TransactionOrders < StubBase
      def list(_account_id, _query = {}, opts: {})
        raise NotImplementedError, 'V2::TransactionOrders#list not wired in 0.1.0'
      end

      def get(_account_id, _id, opts: {})
        raise NotImplementedError, 'V2::TransactionOrders#get not wired in 0.1.0'
      end

      def create(_account_id, _payload, opts: {})
        raise NotImplementedError, 'V2::TransactionOrders#create not wired in 0.1.0'
      end

      def submit(_account_id, _id, opts: {})
        raise NotImplementedError, 'V2::TransactionOrders#submit not wired in 0.1.0'
      end

      def cancel(_account_id, _id, opts: {})
        raise NotImplementedError, 'V2::TransactionOrders#cancel not wired in 0.1.0'
      end
    end

    class Batches < StubBase
      def create(_payload, opts: {})
        raise NotImplementedError, 'V2::Batches#create not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V2::Batches#get not wired in 0.1.0'
      end

      def approve(_id, opts: {})
        raise NotImplementedError, 'V2::Batches#approve not wired in 0.1.0'
      end

      def submit(_id, opts: {})
        raise NotImplementedError, 'V2::Batches#submit not wired in 0.1.0'
      end

      def cancel(_id, opts: {})
        raise NotImplementedError, 'V2::Batches#cancel not wired in 0.1.0'
      end
    end

    class PaymentMethods < StubBase
      def list(_query = {}, opts: {})
        raise NotImplementedError, 'V2::PaymentMethods#list not wired in 0.1.0'
      end

      def get(_id, opts: {})
        raise NotImplementedError, 'V2::PaymentMethods#get not wired in 0.1.0'
      end

      def create(_payload, opts: {})
        raise NotImplementedError, 'V2::PaymentMethods#create not wired in 0.1.0'
      end

      def update(_id, _payload, opts: {})
        raise NotImplementedError, 'V2::PaymentMethods#update not wired in 0.1.0'
      end

      def delete(_id, opts: {})
        raise NotImplementedError, 'V2::PaymentMethods#delete not wired in 0.1.0'
      end
    end

    class Status < StubBase
      def status(opts: {})
        raise NotImplementedError, 'V2::Status#status not wired in 0.1.0'
      end

      def whoami(opts: {})
        raise NotImplementedError, 'V2::Status#whoami not wired in 0.1.0'
      end
    end
  end
end

File: packages/python/src/tesote_sdk/v2/_stubs.py
"""v2 resources not yet wired beyond their public shape."""

from __future__ import annotations

from typing import Any, Dict

from ..transport import Transport


class _StubResource:
    def __init__(self, transport: Transport) -> None:
        self._transport = transport


class TransactionsResource(_StubResource):
    def list_for_account(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 transactions.list_for_account not yet implemented")

    def get(self, transaction_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 transactions.get not yet implemented")

    def export(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 transactions.export not yet implemented")

    def sync(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 transactions.sync not yet implemented")

    def bulk(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 transactions.bulk not yet implemented")

    def search(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 transactions.search not yet implemented")


class SyncSessionsResource(_StubResource):
    def list(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 sync_sessions.list not yet implemented")

    def get(self, account_id: str, session_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 sync_sessions.get not yet implemented")


class TransactionOrdersResource(_StubResource):
    def list(self, account_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 transaction_orders.list not yet implemented")

    def get(self, account_id: str, order_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 transaction_orders.get not yet implemented")

    def create(self, account_id: str, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 transaction_orders.create not yet implemented")

    def submit(self, account_id: str, order_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 transaction_orders.submit not yet implemented")

    def cancel(self, account_id: str, order_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 transaction_orders.cancel not yet implemented")


class BatchesResource(_StubResource):
    def create(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 batches.create not yet implemented")

    def get(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 batches.get not yet implemented")

    def approve(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 batches.approve not yet implemented")

    def submit(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 batches.submit not yet implemented")

    def cancel(self, batch_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 batches.cancel not yet implemented")


class PaymentMethodsResource(_StubResource):
    def list(self) -> Dict[str, Any]:
        raise NotImplementedError("v2 payment_methods.list not yet implemented")

    def get(self, payment_method_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 payment_methods.get not yet implemented")

    def create(self, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 payment_methods.create not yet implemented")

    def update(self, payment_method_id: str, **kwargs: Any) -> Dict[str, Any]:
        raise NotImplementedError("v2 payment_methods.update not yet implemented")

    def delete(self, payment_method_id: str) -> Dict[str, Any]:
        raise NotImplementedError("v2 payment_methods.delete not yet implemented")


class StatusResource(_StubResource):
    def status(self) -> Dict[str, Any]:
        raise NotImplementedError("v2 status not yet implemented")

    def whoami(self) -> Dict[str, Any]:
        raise NotImplementedError("v2 whoami not yet implemented")


__all__ = [
    "TransactionsResource",
    "SyncSessionsResource",
    "TransactionOrdersResource",
    "BatchesResource",
    "PaymentMethodsResource",
    "StatusResource",
]
